src/unix/mod.rs

```diff
@@ -62,17 +62,26 @@ s! {
 
     pub struct timeval {
         pub tv_sec: time_t,
+        #[cfg(not(gnu_time64_abi))]
         pub tv_usec: suseconds_t,
+        // For 64 bit time on 32 bit linux glibc, suseconds_t is still
+        // a 32 bit type.  Using suseconds_t here will break the tests, use i64 instead
+        #[cfg(gnu_time64_abi)]
+        pub tv_usec: i64
     }
 
     // linux x32 compatibility
     // See https://sourceware.org/bugzilla/show_bug.cgi?id=16437
     pub struct timespec {
         pub tv_sec: time_t,
+        #[cfg(all(gnu_time64_abi, any(target_arch = "powerpc", target_arch = "mips")))]
+        __pad: i32,
         #[cfg(all(target_arch = "x86_64", target_pointer_width = "32"))]
```

Comment:
Would it make more sense to use target_endian here, rather than hardcoding specific architectures?

Reply:
Fixed now.